File: dune/xt/grid/test/provider.cc
// This file is part of the dune-xt-grid project:
//   https://github.com/dune-community/dune-xt-grid
// The copyright lies with the authors of this file (see below).
// License: Dual licensed as  BSD 2-Clause License (http://opensource.org/licenses/BSD-2-Clause)
//      or  GPL-2.0+ (http://opensource.org/licenses/gpl-license)
//          with "runtime exception" (http://www.dune-project.org/license.html)
// Authors:
//   Felix Schindler (2012 - 2016)
//   Rene Milk       (2012 - 2016)
//   Tobias Leibner  (2014 - 2016)

#include <dune/xt/common/test/main.hxx>

#include <dune/xt/grid/provider/cube.hh>
#include <dune/xt/grid/provider/dgf.hh>
#include <dune/xt/grid/provider/gmsh.hh>

#include <dune/xt/grid/test/provider.hh>


struct TestGridProvider : public GridProviderBase<Dune::XT::Grid::Providers::TESTGRIDPROVIDERTYPE<TESTGRIDTYPE>>
{
};

TEST_F(TestGridProvider, is_default_creatable)
{
  this->is_default_creatable();
}
TEST_F(TestGridProvider, fulfills_const_interface)
{
  this->const_interface();
}
TEST_F(TestGridProvider, is_visualizable)
{
  this->visualize();
}
TEST_F(TestGridProvider, fulfills_non_const_interface)
{
  this->non_const_interface();
}

File: python/dune/xt/grid/provider.py
# ~~~
# This file is part of the dune-xt project:
#   https://github.com/dune-community/dune-xt
# Copyright 2009-2020 dune-xt developers and contributors. All rights reserved.
# License: Dual licensed as BSD 2-Clause License (http://opensource.org/licenses/BSD-2-Clause)
#      or  GPL-2.0+ (http://opensource.org/licenses/gpl-license)
#          with "runtime exception" (http://www.dune-project.org/license.html)
# Authors:
#   René Fritze    (2018 - 2019)
#   Tobias Leibner (2019 - 2020)
# ~~~

try:
    from dune.xt.grid._provider import *
except ImportError as e:
    import os
    import logging
    if os.environ.get('DXT_PYTHON_DEBUG', False):
        raise e
    logging.error('dune-xt-grid bindings not available')


def default_options_cube_grid(type_str):
    factory = 'GridProviderFactory__{}'.format(type_str)
    try:
        fac = globals()[factory]
    except KeyError:
        raise TypeError('no GridProviderFactory available for Type {}'.format(type_str))
    return fac.default_config('xt.grid.gridprovider.cube')
